morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala

```diff
@@ -3448,6 +3469,57 @@ object EvaluatorMDMTests extends MorphirBaseSpec {
             )
           )
         )
+      ),
+      suite("Morphir Runtime Exception Tests")(
```

Comment:
Are these all the errors you were able to replicate, or is this just a first pass to get the unhappy path machinery set up before flushing out additional errors? (I can help you get a few more runtime errors if that would be helpful, or that can be follow up work)

Reply:
This is what I was able to replicate. There are quite a few exceptions that I feel should be getting thrown but end up throwing a different error, like UnknownTypeMismatch when I would think ArgNumberMismatch would be thrown. If you have consistent ways of replicating other ones, then I'd love to get them added. 

That said, I think adding additional exceptions can be added in separate PRs as needed.

Reply:
One trick you can use is to abuse the type checker:

```
cast : a -> b
cast x = x
```

Morphir-elm (tho not elm) will compile that; you can use it to generate a lot of the other errors. Incomplete pattern matches should be similar (elm will reject but morphir-elm will not). We can talk about other tricks to sneak things past the compiler if that's helpful. You're also completely right, some of these throw other errors than we'd like... a lot of that depends on how the specific error is encountered.

Let's schedule some time to chat soon, I've got a lot of thoughts on these in general.